iothub/device/src/ConnectionStatusChangeReason.cs

```diff
@@ -15,6 +15,7 @@ public enum ConnectionStatusChangeReason
         Retry_Expired,
         No_Network,
         Communication_Error,
-        Client_Close                    
+        Client_Close,
+        Device_Not_Found
     }
```

Comment:
Why can't we throw inner exception for DeviceNotFound ?  Alternately, Shouldn't Device_Disabled be doing the same thing?

Reply:
Not sure I follow the part about 'inner exception'. Edge Agent essentially needs to know when the device is not found in IoT hub anymore. As of now, we were just getting a generic `Communication_Error` `ConnectionStatusChangeReason` when an edge device is deleted from IoT hub instead of something specific such as `Device_Not_Found`.

Regarding `Device_Disabled`, it seems like that was added a while ago and the `DeviceDisabledException` is not really thrown from anywhere.

Reply:
Changed the mapping from `DeviceNotFoundException` to `Device_Disabled` too.